components/templates/ListTodos/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { useMutation } from 'react-query'
+import React, { useCallback, useContext, useEffect } from 'react'
+
+import { TodoContext } from '@/context/TodoContext'
+import { apiRequest } from '@/helpers/common/lib/axios'
+import { TodoType } from '@/helpers/common/types/todoType'
+import TodoItem from './TodoItem'
+import LoadingText from '@/components/common/LoadingText'
+
+type Todos = {
+  todos: TodoType[]
+}
+
+const ListTodos = () => {
+  const { todos, setTodos } = useContext(TodoContext)
+
+  const todosMutationFn = async (): Promise<TodoType[]> => {
+    const { todos } = await apiRequest<Todos>({
+      method: 'POST',
+      url: '/api/todos/get',
+      data: JSON.stringify({}),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+
+    return todos
+  }
+
+  const { mutateAsync: getTodosMutationHandler, isLoading: isLoadingTodos } =
+    useMutation({
```

Comment:
This all works great, but I think if you just used useQuery instead of useMutation, you wouldn't need the context provider. React query already keeps track of your context. Basically, you would have a queryKey for TODOS and react query would do all the caching, context, etc.   

It also keeps track of your loading, fetching, and error handling as well. 

here is an example 

https://tanstack.com/query/v3/docs/react/examples/react/basic